src/state/QuestWrapperProvider.tsx

```diff
@@ -0,0 +1,75 @@
+import { createContext, useContext, ReactNode } from 'react'
+import { TFunction } from 'i18next'
+import { TrackEventFn } from '@/types/analytics'
+import {
+  ConfirmClaimParams,
+  DepositContract,
+  Quest,
+  Reward,
+  RewardClaimSignature,
+  Runner
+} from '@hyperplay/utils'
+
+interface QuestWrapperContextValue {
+  flags: {
+    rewardTypeClaimEnabled: Record<Reward['reward_type'], boolean>
+    questsOverlayClaimCtaEnabled?: boolean
+  }
+  getQuest: (questId: number) => any
+  getUserPlayStreak: (questId: number) => any
+  getSteamGameMetadata: (id: number) => any
+  isSignedIn: boolean
+  trackEvent: TrackEventFn
+  signInWithSteamAccount: () => void
+  openSignInModal: () => void
+  logError: (msg: string) => void
+  claimPoints: (reward: Reward) => Promise<any>
+  completeExternalTask: (reward: Reward) => Promise<any>
+  getQuestRewardSignature: (
+    address: `0x${string}`,
+    rewardId: number,
+    tokenId?: number
+  ) => Promise<RewardClaimSignature>
+  getPendingExternalSync: (questId: number) => Promise<boolean>
+  confirmRewardClaim: (params: ConfirmClaimParams) => Promise<void>
+  syncPlayStreakWithExternalSource: (questId: number) => Promise<unknown>
+  resyncExternalTask: (rewardId: string) => Promise<void>
+  getExternalTaskCredits: (rewardId: string) => Promise<string>
+  syncPlaySession: (appName: string, runner: Runner) => Promise<void>
+  logInfo: (message: string) => void
+  openDiscordLink: () => void
+  getDepositContracts: (questId: number) => Promise<DepositContract[]>
+  tOverride?: TFunction<any, string>
+  sessionEmail?: string
+  checkG7ConnectionStatus: () => Promise<boolean>
+  isQuestsPage?: boolean
+  onPlayClick: (quest: Quest) => void
+  onRewardsClaimed?: (rewards: Reward[]) => void
```

Comment:
this looks to be used only for the quest toast that is shown, first setting the claimed rewards in the `useClaimedRewards` hook https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/1132/files#diff-5005a854a0db0c122c6e1d2fc7ff9c3569aea702bbcb61f5fc915ad1fb781bc7

should we move that hook's state to a separate mobx class that acts as a toast/notification system generally or just for quests? That way all future quest notifications are handled by that mobx class so long as it has been initialized by the consuming code without having to add a new hook and pass a new prop here